src/SWP/Bundle/CoreBundle/Loader/SubscriptionLoader.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Superdesk Web Publisher Core Bundle.
+ *
+ * Copyright 2018 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2018 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Bundle\CoreBundle\Loader;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use SWP\Bundle\CoreBundle\Provider\SubscriptionsProviderInterface;
+use SWP\Component\Paywall\Model\SubscriberInterface;
+use SWP\Component\TemplatesSystem\Gimme\Factory\MetaFactoryInterface;
+use SWP\Component\TemplatesSystem\Gimme\Loader\LoaderInterface;
+use SWP\Component\TemplatesSystem\Gimme\Meta\MetaCollection;
+
+final class SubscriptionLoader implements LoaderInterface
+{
+    public const SUPPORTED_TYPES = ['subscriptions', 'subscription'];
+
+    /**
+     * @var MetaFactoryInterface
+     */
+    private $metaFactory;
+
+    /**
+     * @var SubscriptionsProviderInterface
+     */
+    private $subscriptionsProvider;
+
+    public function __construct(
+        MetaFactoryInterface $metaFactory,
+        SubscriptionsProviderInterface $subscriptionsProvider
+    ) {
+        $this->metaFactory = $metaFactory;
+        $this->subscriptionsProvider = $subscriptionsProvider;
+    }
+
+    public function load($type, $parameters = [], $withoutParameters = [], $responseType = LoaderInterface::SINGLE)
+    {
+        if (!array_key_exists('user', $parameters)) {
+            return false;
+        }
+
+        if (!($user = $parameters['user']) instanceof SubscriberInterface) {
+            return false;
+        }
+
+        $filters = [
+            'routeId' => $parameters['routeId'] ?? null,
+            'articleId' => $parameters['articleId'] ?? null,
+        ];
+
+        if (LoaderInterface::SINGLE === $responseType) {
+            if (!isset($filters['routeId']) && !isset($filters['articleId'])) {
+                return false;
+            }
+
+            $subscription = $this->subscriptionsProvider->getSubscription($user, $filters);
```

Comment:
imo we should provide option to pass subscritpion name here if we look for single one